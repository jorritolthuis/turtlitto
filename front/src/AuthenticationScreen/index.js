import React, { Component } from "react";
import styled from "styled-components";
import PropTypes from "prop-types";

//Move these two to a different location?
import ConnectionBar from "../BottomBar/ConnectionBar";

/**
 * An authentication screen. Upon a submit it will send the token in the
 * inputfield along with a callback function in case the token is incorrect.
 * App.js is responsible for its unrendering.
 * Author: S.A. Tanja
 *
 * Props:
 *  - onSubmit: A function that will send the token to the SRRS
 *  - connectionStatus: The current connection status
 */

class AuthenticationScreen extends Component {
  constructor(props) {
    super(props);
    this.state = {
      token: ""
    };
  }

  render() {
    const { connectionStatus, notification } = this.props;

    return (
      <Container>
        <Window>
          <Label>Token:</Label>
          <Input
            type="text"
            placeholder="Enter the TRC token"
            onChange={event => this.setState({ token: event.target.value })}
          />
          {notification && (
            <WarningLabel id="auth-screen__warning-label">
              {notification}
            </WarningLabel>
          )}
          <LoginButton
            id="login-button"
            onClick={() => this.props.onSubmit(this.state.token)}
          >
            Log in
          </LoginButton>
        </Window>
        <ConnectionWindow>
          <ConnectionBar connectionStatus={connectionStatus} />
        </ConnectionWindow>
      </Container>
    );
  }
}

const Container = styled.div`
  background: ${props => props.theme.loginScreenBackground};
  height: 100%;
  padding-top: 10%;
`;

const Window = styled.div`
  margin: 0 auto;
  border: 0.1rem solid;
  border-bottom: none;
  background: ${props => props.theme.loginFormBackground};
  width: 85%;
  padding: 1rem;
  text-align: center;
`;

const Label = styled.label`
  font-size: 1.5rem;
`;

const WarningLabel = styled(Label)`
  color: ${props => props.theme.error};
`;
WarningLabel.displayName = "WarningLabel";

const Input = styled.input`
  margin-top: 1rem;

  &[type="text"] {
    border: 0.1rem solid;
    width: 100%;
    padding: 1.75rem;
    font-size: 1.5rem;
  }
`;

const LoginButton = styled.button`
  margin-top: 1rem;
  background-color: ${props => props.theme.loginButton};
  border: 0.1rem solid;
  width: 100%;
  height: 5rem;
  font-size: 1.5rem;

  &:active {
    background-color: ${props => props.theme.loginButtonActive};
  }
`;

const ConnectionWindow = styled(Window)`
  padding: 0;
  border-bottom: 0.1rem solid;
`;

AuthenticationScreen.propTypes = {
  onSubmit: PropTypes.func.isRequired,
  notification: PropTypes.string.isRequired
};

export default AuthenticationScreen;
