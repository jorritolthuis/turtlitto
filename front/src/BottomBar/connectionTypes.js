export default Object.freeze({
  CONNECTING: "connecting",
  CONNECTED: "connected",
  DISCONNECTED: "disconnected"
});
