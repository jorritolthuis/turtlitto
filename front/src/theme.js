// these colors are exposed by the styled-components ThemeProvider
// and used within the application
export default {
  success: "#00ff00",
  successLighter: "rgba(153, 255, 153, 0.25)",
  error: "#FF0000",
  errorLighter: "rgba(255,153,153, 0.4)",
  background: "linear-gradient(270deg, #196ebd, #01b0dd)",
  baseButton: "#FFFFFF",
  baseButtonDisabled: "rgba(216,216,216, 0.5)",
  info: "#bdbdbd",
  title: "#FFFFFF",
  secondary: "#d3d3d3",
  border: "#ccc",
  shadow: "rgba(0, 0, 0, 0.17)",
  turtleEnableBar: "#ededed",
  turtleEnableButton: "#ededed",
  turtleEnableButtonActive: "silver",
  bottomBarButton: "#ededed",
  bottomBarButtonActive: "silver",
  notificationSuccess: "green",
  notificationWarning: "#ffcc00",
  notificationError: "Tomato",
  dropBallButton: "#ededed",
  dropBallButtonActive: "silver",
  inOutButton: "#ededed",
  inOutButtonActive: "silver",
  refboxMagenta: "magenta",
  refboxCyan: "cyan",
  refboxMagentaActive: "magenta",
  refboxCyanActive: "cyan"
};
