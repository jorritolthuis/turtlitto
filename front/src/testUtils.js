import React from "react";
import { ThemeProvider } from "styled-components";
import theme from "./theme";
import { shallow, mount } from "enzyme";

export const shallowWithTheme = tree => {
  const context = shallow(<ThemeProvider theme={theme} />)
    .instance()
    .getChildContext();
  return shallow(tree, { context });
};

export const mountWithTheme = tree => {
  const context = shallow(<ThemeProvider theme={theme} />)
    .instance()
    .getChildContext();

  return mount(tree, {
    context,
    childContextTypes: ThemeProvider.childContextTypes
  });
};
