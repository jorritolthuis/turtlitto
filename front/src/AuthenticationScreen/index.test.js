import React from "react";
import AuthenticationScreen from ".";
import connectionTypes from "../BottomBar/connectionTypes";
import { mountWithTheme } from "../testUtils";
import theme from "../theme";

/*
 * Test_items: AuthenticationScreen index.js
 * Input_spec: -
 * Output_spec: -
 * Envir_needs: snapshot (automatically made, found in the __snapshot__ folder).
 */
describe("AuthenticationScreen", () => {
  it("should match snapshot", () => {
    const wrapper = mountWithTheme(
      <AuthenticationScreen
        onSubmit={() => {}}
        connectionStatus={connectionTypes.DISCONNECTED}
        notification={""}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it("should show warning label when notification passed", () => {
    const notificationText = "Incorrect token";
    const wrapper = mountWithTheme(
      <AuthenticationScreen
        onSubmit={() => {}}
        connectionStatus={connectionTypes.DISCONNECTED}
        notification={notificationText}
      />
    );

    const label = wrapper.find("WarningLabel");
    expect(label.text()).toEqual(notificationText);
    expect(label).toHaveStyleRule("color", theme.error);
  });
});
