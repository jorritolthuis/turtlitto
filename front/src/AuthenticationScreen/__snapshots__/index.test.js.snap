// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthenticationScreen should match snapshot 1`] = `
.c6 {
  background: #ff6347;
  padding: 0.6rem 3rem 0.6rem 0.6rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  color: #FFFFFF;
  position: relative;
}

.c7 {
  text-transform: uppercase;
  margin: 0;
  display: block;
  font-size: 1.4rem;
}

.c0 {
  background: #c0c0c0;
  height: 100%;
  padding-top: 10%;
}

.c1 {
  margin: 0 auto;
  border: 0.1rem solid;
  border-bottom: none;
  background: #ededed;
  width: 85%;
  padding: 1rem;
  text-align: center;
}

.c2 {
  font-size: 1.5rem;
}

.c3 {
  margin-top: 1rem;
}

.c3[type="text"] {
  border: 0.1rem solid;
  width: 100%;
  padding: 1.75rem;
  font-size: 1.5rem;
}

.c4 {
  margin-top: 1rem;
  background-color: #FFFFFF;
  border: 0.1rem solid;
  width: 100%;
  height: 5rem;
  font-size: 1.5rem;
}

.c4:active {
  background-color: #ededed;
}

.c5 {
  padding: 0;
  border-bottom: 0.1rem solid;
}

<AuthenticationScreen
  connectionStatus="disconnected"
  notification=""
  onSubmit={[Function]}
>
  <styled.div>
    <div
      className="c0"
    >
      <styled.div>
        <div
          className="c1"
        >
          <styled.label>
            <label
              className="c2"
            >
              Token:
            </label>
          </styled.label>
          <TokenInput
            id="auth-screen_token-input"
            onChange={[Function]}
            placeholder="Enter the TRC token"
            type="text"
          >
            <input
              className="c3"
              id="auth-screen_token-input"
              onChange={[Function]}
              placeholder="Enter the TRC token"
              type="text"
            />
          </TokenInput>
          <LoginButton
            id="auth-screen__login-button"
            onClick={[Function]}
          >
            <button
              className="c4"
              id="auth-screen__login-button"
              onClick={[Function]}
            >
              Log in
            </button>
          </LoginButton>
        </div>
      </styled.div>
      <Styled(styled.div)>
        <styled.div
          className="c5"
        >
          <div
            className="c5 c1"
          >
            <ConnectionBar
              connectionStatus="disconnected"
            >
              <styled.div
                background={
                  Array [
                    "
                          background: ",
                    [Function],
                    ";
                        ",
                  ]
                }
              >
                <div
                  className="c6"
                >
                  <styled.p>
                    <p
                      className="c7"
                    >
                      disconnected
                    </p>
                  </styled.p>
                </div>
              </styled.div>
            </ConnectionBar>
          </div>
        </styled.div>
      </Styled(styled.div)>
    </div>
  </styled.div>
</AuthenticationScreen>
`;
